tensorboard/webapp/metrics/views/card_renderer/utils_test.ts

```diff
@@ -310,4 +311,140 @@ describe('metrics card_renderer utils test', () => {
       expect(getClosestStep(10, [0, 10, 20])).toBe(10);
     });
   });
+
+  describe('#maybeClipLinkedTimeSelection', () => {
+    it('changes start when start.step < minStep', () => {
+      expect(
+        maybeClipLinkedTimeSelection(
+          {
+            start: {step: 0},
+            end: null,
+          },
+          1,
+          4
+        )
+      ).toEqual({
+        startStep: 1,
+        endStep: null,
+        clipped: true,
+      });
+    });
+
+    it('changes end when end.step < maxStep', () => {
```

Comment:
ditto: something like "clips to maxStep when time selection end step is greater than view extend" ?